src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -169,7 +170,7 @@ static int parse_efi_hibernate_location(void) {
                 arg_resume_device = TAKE_PTR(device);
                 arg_resume_offset = location.offset;
         } else {
-                if (!streq(arg_resume_device, device))
+                if (!path_equal_or_inode_same(arg_resume_device, device, 0))
```

Comment:
this isn't fully right. this is a device node, hence it you want to compare the device major/minor of the inode, but not the inode itself.

i.e. path_requal_or_inode_same() will compare path, or the `.st_ino` and `.st_dev` fields of `struct stat`. But here we should probably compare `.st_rdev` instead, because two device nodes referencing the same backing device should be OK even if they entrypoint device node inodes are different.

maybe add a helper wrapper: 

```c
static int path_equal_or_inode_or_devnode_same(const char *a, const char *b, int flags) {
        struct stat sa, sb;  
        if (path_equal(a, b))
                return true;
        if (fstatat(AT_FDCWD, a, &sa, flags) < 0)
                return -errno;
        if (fstatat(AT_FCCWD, b, &sb, flags) < 0)
                return -errno;
        if (stat_inode_same(&a, &sb))
                return true;
        if (!S_ISBLK(sa.st_mode) || !S_ISCHR(sb.st_mode)) /* char or block device node? */
                return false;
        if (((sa->st_mode ^ sb->st_mode) & S_IFMT) != 0) /* both inode same device node type? */
                return false;
        return sa->st_rdev == sb->st_rdev; /* referencing same device? */
}
```

Reply:
-> #28344